src/main/scala/za/co/absa/hyperdrive/trigger/persistance/DagRunRepository.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.trigger.persistance
+
+import org.springframework.stereotype
+import slick.dbio.Effect
+import slick.sql.{FixedSqlAction, FixedSqlStreamingAction}
+import za.co.absa.hyperdrive.trigger.models.dagRuns.{DagRun, DagRunsSearchRequest, DagRunsSearchResponse, Filters, RangeFilters}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait DagRunRepository extends Repository {
```

Comment:
There should be a test here as well :) However, given that you don't call this code currently with filters and sorts (correct me if I'm wrong), the code is probably going to change anyway, so I'm ok with postponing the test for now.

Reply:
I have implemented some tests. I dont test filter because we can expect changes there.